### PR TITLE
Attempt to improve publish-docs.yml

### PR DESCRIPTION
**Pull request by ahmdhjj**

### CODE REVIEW

**Comment by ahmdhjj** [comment]
@naz-hage In [workflow_run](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run), there is a note that it will only trigger a workflow run if the workflow file is on the default branch. This is not suitable for us, I think we can combine the two jobs (build and docs) in one workflow file and that will cause them to chain, see screenshot below (I'll work on fixing the error in docs)

Screenshot 2024-04-27 at 12 40 47 PM

File: .github/workflows/dotnet-desktop.yml [hunk 1]
```diff
@@ -160,3 +155,23 @@ jobs:
...
+        with:
+          python-version: '3.12'
+      - name: Install dependencies  
+        run: |  
+          python -m pip install --upgrade pip  
+          pip install mkdocs  
+      - name: Update docs  
+        run: |  
+          mkdocs build  
+          mkdocs gh-deploy --force
```
Comment (naz-hage) [inline .github/workflows/dotnet-desktop.yml:177]: Possibly you need to setup the working space

**Comment by naz-hage** [comment]
If this is the case, why not use windows latest runner? Why spawn another Ubuntu runner just for docs?

**Comment by ahmdhjj** [comment]
> If this is the case, why not use windows latest runner? Why spawn another Ubuntu runner just for docs?

My understanding is that runners are restricted to individual jobs. If we merge the two jobs (adding docs steps in the build job), that would work, the docs will also depend on the steps before but we wouldn't have a chain.

File: .github/workflows/dotnet-desktop.yml (hunk 1 above)
Comment (ahmdhjj) [reply .github/workflows/dotnet-desktop.yml:175]: Should be fixed now

**Comment by naz-hage** [comment]
> > If this is the case, why not use windows latest runner? Why spawn another Ubuntu runner just for docs?
> 
> My understanding is that runners are restricted to individual jobs. If we merge the two jobs (adding docs steps in the build job), that would work, the docs will also depend on the steps before but we wouldn't have a chain.

You're right, Yes, each job in a GitHub workflow runs on a runner.

**Review by naz-hage — APPROVED** [review]

**Pull request merged**